sdk/storage/Azure.Storage.DataMovement/tests/Shared/DataMovementTestBase.cs

```diff
@@ -74,5 +79,193 @@ public List<string> ListFilesInDirectory(string directory)
             }
             return files;
         }
+
+        public static DisposingLocalDirectory GetTestLocalDirectory(string directoryPath = default)
```

Comment:
oh I was just pattern matching what we had in blobs for the disposing container. we should probably make a work item to remove that too since I was unaware about this issue.

Also what specific helper classes are we talking about here? Like adding this to each derived test base? What's the suggestion of what to replace this with?

Reply:
The suggestion is to have an API like this be either:
1. on a static helper class, assuming we don't need access to things handled by existing base class functionality
2. on an instanced helper class that takes in the appropriate level of base class as an instance via passing `this` in the constructor (e.g. public TestDirectoryManager(RecordedTestBase testBase)`).

The end goal is to not have DataMovementTestBase at all. TestBase classes I've found often get in the way when trying to modularize test components and end up needing to get refactored out. Additionally, often this sort of functionality ends up being copy/pasted from some other package's TestBase. Having a shared modular component leads to our tests being consistent in their behavior, ensuring our code that is conceptually the same actually behaves the same.

Getting rid of the base class isnt' a requirement for this PR but we should not be adding more functionality that ends up needing refactoring out in some future.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/35408